packages/ember-htmlbars/lib/hooks/component.js

```diff
@@ -33,6 +34,29 @@ export default function componentHook(renderNode, env, scope, _tagName, params,
     let componentCell = stream.value();
     if (isComponentCell(componentCell)) {
       tagName = componentCell[COMPONENT_PATH];
+
+      /*
+       * This need to be done to avoid problems with rest positional parameters
+       * when rendered using the dot notation.
+       *
+       * Closure components (for the contextual component feature) does not
+       * actually keep the positional params, but process them at each level.
+       * Therefore, when rendering a closure component with the component
+       * helper we just process the parameters and attributes and then we
```

Comment:
"we process the parameters and attributes and then merge those on top of the closure component attributes"